app/questionnaire/questionnaire_store_updater.py

```diff
@@ -505,6 +439,11 @@ def _evaluate_dependent_of_dependents(
         )
         evaluated_dependents.append((dependent_section_id, list_item_id))
 
+    def is_current_section(self, section_id: str, list_item_id: str | None) -> bool:
```

Comment:
Do we need to pass in arguments if this method doesn't do anything? Shouldn't we create another definition of `remove_dependent_blocks_and_capture_dependent_sections` in the child class that actually makes correct use of `is_current_section`?

Reply:
This class' `remove_dependent_blocks_and_capture_dependent_sections` could be?:
```python
def remove_dependent_blocks_and_capture_dependent_sections(self) -> None:
        """Removes dependent blocks from the progress store."""
                    list_item_id=list_item_id,
                    block_id=block_id,
                )
                blocks_removed |= self._progress_store.remove_completed_location(
                    location
                )

            if blocks_removed:
 ```

Reply:
Yeah, I prefer your latest change, that method before, always returning `False` and all these ignores, was sort of like a `code smell`, hard to explain.

Reply:
Yeah I felt similarly about it and couldn't think of a better way at the time, but I agree so good to get rid of it, thanks for flagging 👍